portal/date_tools.py

```diff
@@ -172,3 +172,21 @@ def utcnow_sans_micro():
     """
     now = datetime.utcnow()
     return now.replace(microsecond=0)
+
+
+def weekday_delta(start, end):
```

Comment:
Will this function mutate `start`? Is that a problem?

Reply:
reasonable concern, but i don't believe it's modifying `start` or `end` - that's the purpose of using `i` in this loop as the mutable form.